Moosh/Command/Moodle23/File/DBCheck.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * moosh - Moodle Shell
+ *
+ * @copyright  2012 onwards Tomasz Muras
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * DBCheck - Darío Gómez <dario.gomez@solutionsad.com>
+ */
+
+namespace Moosh\Command\Moodle23\File;
+
+use Moosh\MooshCommand;
+
+class DBCheck extends MooshCommand {
+    private $quiet = false;
+    
+    public function __construct() {
+        parent::__construct('dbcheck', 'file');
+        $this->addOption('q|quiet', 'Quiet mode. Only outputs errors.');
+    }
+
+    protected function getArgumentsHelp() {
+        $help = parent::getArgumentsHelp();
+        $help .= "\n\n";
+        $help .= "Check moodledata files for coruption.";
+
+        return $help;
+    }
+
+    private function checkDirectory($dir, &$errors) {
+        $dir .= DIRECTORY_SEPARATOR;
+        
+        // Read contents of current directory
+        $list = scandir($dir);
+        
+        // Exclude .  ..  warning.txt
+        $list = array_diff($list, array('.', '..', 'warning.txt'));
+        
+        foreach ($list as &$entry) {
+            if (is_dir($dir.$entry)) {
+                // Recursive call on directories
+                $this->checkDirectory($dir.$entry, $errors);
+            }
+            
+            else {
+                if (!$this->checkFile($entry)) {
+                    $errors[] = $dir.$entry;
+                }
+            }
+            
+            if (!$this->quiet) echo '.';
+        }
+    }
+    
+    private function checkFile($file) {
+	global $DB;
+	
+	$sql = "SELECT id
```

Comment:
checkFile function is used on line 46. This function test a file hash is present into database.